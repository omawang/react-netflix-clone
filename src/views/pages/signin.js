import React, { useContext, useState } from 'react';
import { useHistory } from 'react-router-dom';

import { FirebaseContext } from '../../contexts/firebase';
import { Form } from '../components';
import { FooterContainer } from '../containers/footer';
import { HeaderContainer } from '../containers/header';

import * as ROUTES from '../../constants/routes';

export default function Signin() {
  const history = useHistory();
  const { firebase } = useContext(FirebaseContext);

  const [emailAddress, setEmailAddress] = useState('');
  const [password, setPassword] = useState('');
  const [error, setError] = useState('');

  const isInvalid = emailAddress === '' || password === '';

  const handleSignIn = (e) => {
    e.preventDefault();
    firebase
      .auth()
      .signInWithEmailAndPassword(emailAddress, password)
      .then(() => {
        history.push(ROUTES.BROWSE);
      })
      .catch((error) => {
        setEmailAddress('');
        setPassword('');
        setError(error.message);
      });
  };

  return (
    <>
      <HeaderContainer>
        <Form>
          <Form.Title>Sign In Now</Form.Title>
          {error && <Form.Error data-testid="error">{error}</Form.Error>}

          <Form.Base method="POST" onSubmit={handleSignIn}>
            <Form.Input
              placeholder="Email address"
              value={emailAddress}
              onChange={({ target }) => setEmailAddress(target.value)}
            />

            <Form.Input
              placeholder="Password"
              type="password"
              autoComplete="off"
              value={password}
              onChange={({ target }) => setPassword(target.value)}
            />

            <Form.Submit
              disabled={isInvalid}
              type="submit"
              data-testid="signin-submit"
            >
              Sign In
            </Form.Submit>
          </Form.Base>

          <Form.Text>
            New to Netflix?{' '}
            <Form.Link to={ROUTES.SIGN_UP}>Sign up now.</Form.Link>
          </Form.Text>
          <Form.TextSmall>
            This page is protected by Google reCAPTCHA to ensure you&apos;re not
            a bot. Learn more.
          </Form.TextSmall>
        </Form>
      </HeaderContainer>

      <FooterContainer />
    </>
  );
}
