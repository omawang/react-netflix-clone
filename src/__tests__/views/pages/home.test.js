import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import { Home } from '../../../views/pages';

describe('Homepage', () => {
  it('render Home page', () => {
    const { container, getByText, getByTestId } = render(
      <Router>
        <Home />
      </Router>
    );

    expect(getByText('Unlimited films, TV programmes and more.')).toBeTruthy();
    expect(getByText('Watch anywhere. Cancel at any time.')).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
