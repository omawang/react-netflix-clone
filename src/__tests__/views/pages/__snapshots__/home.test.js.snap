// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage render Home page 1`] = `
<div
  class="header__Background-sc-1v45rsj-0 gvksrW"
>
  <div
    class="header__Container-sc-1v45rsj-1 eSeMKt"
  >
    <a
      href="/"
    >
      <img
        alt="Netflix"
        class="header__Logo-sc-1v45rsj-6 kbpjKE"
        data-testid="header-bg"
        src="logo.svg"
      />
    </a>
    <a
      class="header__ButtonLink-sc-1v45rsj-7 jNQYQi"
      href="/signin"
    >
      Sign In
    </a>
  </div>
  <div
    class="feature__Container-sc-17lpqa4-0 hDwihu"
  >
    <h1
      class="feature__Title-sc-17lpqa4-1 eLQpaR"
    >
      Unlimited films, TV programmes and more.
    </h1>
    <h2
      class="feature__SubTitle-sc-17lpqa4-2 gEmVAy"
    >
      Watch anywhere. Cancel at any time.
    </h2>
    <div
      class="opt-form__Container-z45k4t-0 kLhZhH"
    >
      <input
        class="opt-form__Input-z45k4t-1 kskwzu"
        placeholder="Email address"
      />
      <button
        class="opt-form__Button-z45k4t-2 iOtied"
      >
        Try it now
         
        <img
          alt="Try now"
          src="/images/icons/chevron-right.png"
        />
      </button>
      <div
        class="opt-form__Break-z45k4t-4 fVIGjV"
      />
      <p
        class="opt-form__Text-z45k4t-3 bqXIIx"
      >
        Ready to watch? Enter your email to create or restart your membership.
      </p>
    </div>
  </div>
</div>
`;
